OpenSim/Tools/ModelScaler.cpp

```diff
@@ -311,27 +311,30 @@ bool ModelScaler::processModel(Model* aModel, const string& aPathToSubject,
         /* Now scale the model. */
         aModel->scale(s, theScaleSet, aSubjectMass, _preserveMassDist);
 
-
         if(_printResultFiles) {
             std::string savedCwd = IO::getCwd();
             IO::chDir(aPathToSubject);
+            try { // writing can throw an exception
+                if (_outputModelFileNameProp.isValidFileName()) {
+                    if (aModel->print(_outputModelFileName))
+                        cout << "Wrote model file " << _outputModelFileName <<
+                        " from model " << aModel->getName() << endl;
+                }
 
-            if (!_outputModelFileNameProp.getValueIsDefault())
-            {
-                if (aModel->print(_outputModelFileName))
-                    cout << "Wrote model file " << _outputModelFileName << " from model " << aModel->getName() << endl;
+                if (_outputScaleFileNameProp.isValidFileName()) {
+                    if (theScaleSet.print(_outputScaleFileName))
+                        cout << "Wrote scale file " << _outputScaleFileName <<
+                        " for model " << aModel->getName() << endl;
+                }
             }
-
-            if (!_outputScaleFileNameProp.getValueIsDefault())
-            {
-                if (theScaleSet.print(_outputScaleFileName))
-                    cout << "Wrote scale file " << _outputScaleFileName << " for model " << aModel->getName() << endl;
+            catch (std::exception& ex) {
```

Comment:
The comment in the analogous place in MarkerPlacer.cpp (`// catch the exception so we can reset the working directory`) would have the same usefulness here.